### DISCUSSION TITLE
HawkEye Reports Throwing Issue On Staging Environment

### DISCUSSION THREAD

**Discussion by Shakthieshwari** [Sunbird-Obsrv/Community#11] [Issues]
Dear Team,

In staging , We created a hawkeye report.

job id :- ml_total_submissions_new_api_new_testc

{
    "id": "ekstep.analytics.report.get",
    "ver": "1.0",
    "ts": "2022-06-02T14:10:47.583+00:00",
    "params": {
        "resmsgid": "d13fc227-1368-4cca-b0cb-d2478aeb7e7a",
        "status": "successful",
        "client_key": null
    },
    "responseCode": "OK",
    "result": {
        "createdOn": 1653943478527,
        "updatedOn": 1654162767802,
        "reportId": "ml_total_submissions_new_api_new_testc",
        "config": {
            "streamQuery": true,
            "reportConfig": {
                "id": "ml_total_submissions_new_api_new_testc",
                "mergeConfig": {
                    "postContainer": "reports",
                    "container": "reports",
                    "basePath": "/mount/data/analytics/tmp",
                    "reportPath": "ml_total_submissions_new_api_new_testc.csv",
                    "rollup": 0
                },
                "labels": {
                    "SUM(unique_submissions)": "total_content_plays_on_portal",
                    "legend": "Total submissions",
                    "total_content_plays_on_portal": "Total submissions",
                    "organisation_name": "Organisation",
                    "parent_channel": "parent_channel",
                    "district_name": "District name",
                    "program_name": "Program name",
                    "state_slug": "state_slug",
                    "solution_name": "Observation name"
                },
                "dateRange": {
                    "staticInterval": "LastDay",
                    "granularity": "all",
                    "intervalSlider": 0
                },
                "metrics": [
                    {
                        "metric": "total_content_plays_on_portal",
                        "label": "total_content_plays_on_portal",
                        "druidQuery": {
                            "dataSource": "ml-distinctCnt-obs-status",
                            "limitSpec": {
                                "type": "default",
                                "limit": 10000,
                                "columns": [
                                    {
                                        "dimension": "SUM(unique_submissions)",
                                        "direction": "descending"
                                    }
                                ]
                            },
                            "filters": [
                                {
                                    "type": "equals",
                                    "dimension": "status",
                                    "value": "completed"
                                },
                                {
                                    "type": "equals",
                                    "dimension": "private_program",
                                    "value": "false"
                                },
                                {
                                    "type": "equals",
                                    "dimension": "solution_type",
                                    "value": "observation_with_out_rubric"
                                }
                            ],
                            "granularity": "all",
                            "dimensions": [
                                {
                                    "fieldName": "parent_channel",
                                    "aliasName": "parent_channel"
                                },
                                {
                                    "type": "extraction",
                                    "extractionFn": [
                                        {
                                            "type": "registeredLookup",
                                            "retainMissingValue": true,
                                            "fn": "stateSlugLookup"
                                        }
                                    ],
                                    "fieldName": "state_name",
                                    "aliasName": "state_slug"
                                },
                                {
                                    "fieldName": "organisation_name",
                                    "aliasName": "Organisation"
                                },
                                {
                                    "fieldName": "district_name",
                                    "aliasName": "District name"
                                },
                                {
                                    "fieldName": "program_name",
                                    "aliasName": "Program name"
                                },
                                {
                                    "fieldName": "solution_name",
                                    "aliasName": "Observation name"
                                }
                            ],
                            "aggregations": [
                                {
                                    "fieldName": "unique_submissions",
                                    "type": "longSum",
                                    "name": "total_content_plays_on_portal"
                                }
                            ],
                            "postAggregations": [],
                            "queryType": "groupBy"
                        }
                    }
                ],
                "output": [
                    {
                        "type": "csv",
                        "metrics": [
                            "total_content_plays_on_portal"
                        ],
                        "dims": [
                            "date",
                            "state_slug"
                        ],
                        "fileParameters": [
                            "id",
                            "dims"
                        ]
                    }
                ],
                "queryType": "groupBy"
            },
            "container": "reports",
            "store": "azure",
            "key": "hawk-eye/"
        },
        "requestedBy": "fca2925f-1eee-4654-9177-fece3fd6afc9",
        "reportDescription": "Total no of unique entities for which observation are done",
        "submittedOn": 1653943478527,
        "status": "ACTIVE",
        "status_msg": "REPORT SUCCESSFULLY ACTIVATED",
        "reportSchedule": "ONCE"
    }
}

We are getting this error :-
**java.util.NoSuchElementException: key not found: reportConfig**

reportConfig key is present, but even though it is showing the error. Can you please help us out ?

Thanks

Cc- @aks30, @kiranharidas187, @vijiurs

**Comment by Shakthieshwari** [comment]
Team, Any update on this please ? Its a blocker for us to create the reports in staging

Thanks

**Comment by Shakthieshwari** [comment]
Team, Any update on this please ? Its a blocker for us to create the reports in staging

Thanks

Cc- @aks30, @kiranharidas187, @vijiurs

**Reply by sowmya-dixit** [reply]
@Shakthieshwari  - There was some Jenkins job issue. Now it's resolved. Can you try re-running the report once and let us know if that works?

**Reply by Shakthieshwari** [reply]
Working fine now. thanks @sowmya-dixit
